parcels/particleset/baseparticleset.py

```diff
@@ -433,12 +446,34 @@ def execute(self, pyfunc=AdvectionRK4, endtime=None, runtime=None, dt=1.,
                                 'to a NetCDF file during the run.' % output_file.tempwritedir_base)
                 pbar = self.__create_progressbar(_starttime, endtime)
                 verbose_progress = True
+
             if dt > 0:
-                time = min(next_prelease, next_input, next_output, next_movie, next_callback, endtime)
+                next_time = min(next_prelease, next_input, next_output, next_movie, next_callback, endtime)
+            else:
+                next_time = max(next_prelease, next_input, next_output, next_movie, next_callback, endtime)
+
+            # qubixes: I am not sure if this is how it is supposed to work..
```

Comment:
@CKehl @erikvansebille Can you have a look at this section of the code and see if this is correct for all cases?

Reply:
see comment below, please